api/v2/log.go

```diff
@@ -12,24 +12,25 @@ import (
 // is the value
 func (api *API) LogError(err error, message string, fields ...interface{}) func(c *gin.Context, code ...int) {
 	// create base entry
-	entry := api.l.WithFields(log.Fields{
-		"service": api.service,
-	})
-
-	// add additional fields
-	if fields != nil {
-		for i := 0; i < len(fields); i += 2 {
-			if i+1 < len(fields) {
-				entry = entry.WithField(fields[i].(string), fields[i+1])
-			}
-		}
-	}
+	logger := api.l
 
 	// write log
 	if err == nil {
-		entry.Error(message)
+		// mod 2 allows us to check the fields are even in length
+		if fields != nil && len(fields)%2 == 0 {
+			logger.With(fields...).Error(message)
+		} else {
+			logger.Error(message)
+		}
 	} else {
-		entry.WithField("error", err.Error()).Error(message)
+		// mod 2 allows us to check that fields are even in length
+		if fields != nil && len(fields)%2 == 0 {
+			fields = append(fields, "error")
+			fields = append(fields, err.Error())
+			logger.With(fields...).Error(message)
+		} else {
+			logger.With("error", err.Error()).Error(message)
```

Comment:
same here, remove this whole chunk and just do a single `logger.Errorw(message, "error", err.Error())`